### ISSUE TITLE
Gateway crashed upon closing tab

### ISSUE THREAD

**Issue by mrstegeman** [mozilla-iot/gateway#743]
```
2018-03-07 17:55:22.532 /home/pi/mozilla-iot/gateway/node_modules/express-ws/node_modules/ws/lib/WebSocket.js:311
      throw new Error('not opened');
            ^
2018-03-07 17:55:22.533 Error: not opened
    at WebSocket.ping (/home/pi/mozilla-iot/gateway/node_modules/express-ws/node_modules/ws/lib/WebSocket.js:311:13)
    at Timeout._onTimeout (/home/pi/mozilla-iot/gateway/build/webpack:/src/controllers/things_controller.js:215:1)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)

npm ERR! Linux 4.9.59-v7+
npm ERR! argv "/home/pi/.nvm/versions/node/v7.10.1/bin/node" "/home/pi/.nvm/versions/node/v7.10.1/bin/npm" "start"
npm ERR! node v7.10.1
npm ERR! npm  v4.2.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! things-gateway@0.3.1 start: `webpack && node build/gateway.js`
npm ERR! Exit status 1
```

**Comment by hobinjk** [comment]
This looks like a race between `cleanup` (called when the close event is emitted by the websocket), and the actual closing of the websocket. I'm not really sure how to deal with it.

**Comment by mrstegeman** [comment]
I think the appropriate thing to do is have the systemd service restart the gateway on crash. Currently, it's set to roll back an update, which is maybe not the most appropriate action to take in all cases.

**Comment by hobinjk** [comment]
It only rolls back if the update happened within the past two days (possibly two weeks).

**Comment by mrstegeman** [comment]
Gateway race condition is exceptionally rare. The errant update rollback has been fixed here: #744
